Tools-Override/Build.Common.props

```diff
@@ -23,10 +23,12 @@
 
   <!-- Platform detection -->
   <PropertyGroup>
-    <OsEnvironment Condition="'$(OsEnvironment)'=='' AND '$(OS)'=='Unix' AND Exists('/Applications')">OSX</OsEnvironment>
-    <OsEnvironment Condition="'$(OsEnvironment)'=='' AND '$(OS)'=='Unix'">Linux</OsEnvironment>
-    <OsEnvironment Condition="'$(OsEnvironment)'==''">$(OS)</OsEnvironment>
```

Comment:
@mellinoe noticed this while looking at the other problem: there are still a few tasks in buildtools that rely on OsEnvironment being set, e.g.: https://github.com/dotnet/buildtools/search?utf8=%E2%9C%93&q=OsEnvironment

I think with this change those would be doing the wrong thing.